cpp/tests/c_api/uniform_neighbor_sample_test.c

```diff
@@ -0,0 +1,203 @@
+                                         int* h_start_label,
+                                         size_t num_starts,
+                                         int* fan_out,
+                                         size_t max_depth,
+                                         bool_t with_replacement,
+                                         bool_t renumber,
+                                         bool_t store_transposed)
+{
+  int test_ret_value = 0;
+
+  cugraph_error_code_t ret_code = CUGRAPH_SUCCESS;
+  cugraph_error_t* ret_error    = NULL;
+
+  cugraph_resource_handle_t* handle = NULL;
+  cugraph_graph_t* graph            = NULL;
+  cugraph_sample_result_t* result   = NULL;
+
+  cugraph_type_erased_device_array_t* d_start                 = NULL;
+  cugraph_type_erased_device_array_view_t* d_start_view       = NULL;
+  cugraph_type_erased_device_array_t* d_start_label           = NULL;
+  cugraph_type_erased_device_array_view_t* d_start_label_view = NULL;
+  cugraph_type_erased_host_array_view_t* h_fan_out_view       = NULL;
+
+  handle = cugraph_create_resource_handle(NULL);
+  TEST_ASSERT(test_ret_value, handle != NULL, "resource handle creation failed.");
+
+  ret_code = create_test_graph(
+    handle, h_src, h_dst, h_wgt, num_edges, store_transposed, renumber, &graph, &ret_error);
+  TEST_ASSERT(test_ret_value, ret_code == CUGRAPH_SUCCESS, "graph creation failed.");
+
+  ret_code =
+    cugraph_type_erased_device_array_create(handle, num_starts, INT32, &d_start, &ret_error);
+  TEST_ASSERT(test_ret_value, ret_code == CUGRAPH_SUCCESS, "d_start create failed.");
+
+  d_start_view = cugraph_type_erased_device_array_view(d_start);
+
+  ret_code =
+    cugraph_type_erased_device_array_create(handle, num_starts, INT32, &d_start_label, &ret_error);
+  TEST_ASSERT(test_ret_value, ret_code == CUGRAPH_SUCCESS, "d_start_label create failed.");
+
+  d_start_label_view = cugraph_type_erased_device_array_view(d_start_label);
+
+  ret_code = cugraph_type_erased_device_array_view_copy_from_host(
+    handle, d_start_label_view, (byte_t*)h_start_label, &ret_error);
+  TEST_ASSERT(test_ret_value, ret_code == CUGRAPH_SUCCESS, "start copy_from_host failed.");
+
+  h_fan_out_view = cugraph_type_erased_host_array_view_create(fan_out, max_depth, INT32);
+
+  ret_code = cugraph_uniform_neighbor_sample(handle,
+                                             graph,
+                                             d_start_view,
+                                             d_start_label_view,
+                                             h_fan_out_view,
+                                             with_replacement,
+                                             FALSE,
+                                             &result,
+                                             &ret_error);
+
+  TEST_ASSERT(test_ret_value,
+              ret_code != CUGRAPH_SUCCESS,
+              "cugraph_uniform_neighbor_sample expected to fail in SG test");
+
+#if 0
```

Comment:
Should we keep this?

Reply:
I'm hoping to add support for SG in the next release.  This should be the code to do the test once SG is working (although it will need to be debugged).  I can delete it and recreate it from the MG version next release if we want to keep this clean, although I'm inclined to keep it for now.

Reply:
Sure, no problem.